test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -3,240 +3,234 @@
     {
-        private readonly TestServer _testServer;
-        private static readonly bool _verbose = string.Equals(Environment.GetEnvironmentVariable("SIGNALR_TEST_VERBOSE"), "1");
+        public const string Name = "EndToEndTests";
+    }
 
-        public HubConnectionTests()
+    [Collection(HubConnectionTestsCollection.Name)]
+    public class HubConnectionTests : LoggedTest
+    {
+        private readonly ServerFixture _serverFixture;
+        public HubConnectionTests(ServerFixture serverFixture, ITestOutputHelper output)
+            : base(output)
         {
-            var webHostBuilder = new WebHostBuilder().
-                ConfigureServices(services =>
-                {
-                    services.AddSignalR();
-                })
-                .ConfigureLogging(loggerFactory =>
-                {
-                    if (_verbose)
-                    {
-                        loggerFactory.AddConsole();
-                    }
-                })
-                .Configure(app =>
-                {
-                    app.UseSignalR(routes =>
-                    {
-                        routes.MapHub<TestHub>("hubs");
-                    });
-                });
-            _testServer = new TestServer(webHostBuilder);
+            if (serverFixture == null)
+            {
+                throw new ArgumentNullException(nameof(serverFixture));
+            }
+
+            _serverFixture = serverFixture;
         }
 
         [Theory]
-        [MemberData(nameof(HubProtocols))]
-        public async Task CheckFixedMessage(IHubProtocol protocol)
+        [MemberData(nameof(HubProtocolsXTransports))]
+        public async Task CheckFixedMessage(IHubProtocol protocol, TransportType transport)
         {
-            var loggerFactory = CreateLogger();
-
-            var httpConnection = new HttpConnection(new Uri("http://test/hubs"), TransportType.LongPolling, loggerFactory, _testServer.CreateHandler());
-            var connection = new HubConnection(httpConnection, protocol, loggerFactory);
-            try
+            using (StartLog(out var loggerFactory))
             {
-                await connection.StartAsync();
+                var httpConnection = new HttpConnection(new Uri(_serverFixture.BaseUrl + "/hubs"), transport, loggerFactory);
+                var connection = new HubConnection(httpConnection, protocol, loggerFactory);
+                try
+                {
+                    await connection.StartAsync().OrTimeout();
 
-                var result = await connection.InvokeAsync<string>(nameof(TestHub.HelloWorld));
+                    var result = await connection.InvokeAsync<string>(nameof(TestHub.HelloWorld)).OrTimeout();
 
-                Assert.Equal("Hello World!", result);
-            }
-            finally
-            {
-                await connection.DisposeAsync();
+                    Assert.Equal("Hello World!", result);
+                }
+                finally
```

Comment:
Add something like:
```
catch (Exception ex)
{
    loggerFactory.CreateLogger("Test").LogError(ex, "Exception from test");
    throw;
}
```

Not having this can hide test issues if `await connection.DisposeAsync().OrTimeout();` ends up throwing after an exception was already thrown